src/app/metric/google-sheets/metric-config.tsx

```
@@ -0,0 +1,309 @@
+
+  // Fetch sheets for dropdown (depends on SPREADSHEET_ID)
+  const shouldFetchSheets = sheetParam?.dynamicConfig && params.SPREADSHEET_ID;
+
+  const sheetEndpoint =
+    sheetParam?.dynamicConfig?.endpoint?.replace(
+      "{SPREADSHEET_ID}",
+      params.SPREADSHEET_ID ?? "",
+    ) ?? "";
+
+  const { data: sheetsData, isLoading: isLoadingSheets } =
+    api.metric.fetchIntegrationData.useQuery(
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

Hooks should not be called after an early return.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)